src/YourEasyRent/UserState/UserSearchState.cs

```diff
@@ -0,0 +1,115 @@
+        public IEnumerable<MenuStatus> HistoryOfMenuStatuses
+        {
+            get { return _historyOfMenuStatuses; } 
+            private set { _historyOfMenuStatuses = (List<MenuStatus>)value; }
+        }
+
+        private List<MenuStatus> _historyOfMenuStatuses = new List<MenuStatus>();
+
+        public UserSearchState()
+        {
+            //UserId = userId.ToString();
+            //CurrentMenuStatus = MenuStatus.Started;
+            //_historyOfMenuStatuses.Add(MenuStatus.Started);
+
+        }
+
+        public UserSearchState(UserSearchStateDTO dto)
+        {
+            UserId = dto.UserId;
+            ChatId = dto.ChatId;
+            Category = dto.Category;
+            Brand = dto.Brand;
+            CurrentMenuStatus = dto.Status;
+            _historyOfMenuStatuses = dto.HistoryOfMenuStatuses;
+
+        }
+        public UserSearchState CreateNewUserSearchState(string userId) // factory
+        {
+            UserSearchState userSearchState = new UserSearchState
+            {
+                UserId = userId,
+                HistoryOfMenuStatuses = new List<MenuStatus>(),
+                CurrentMenuStatus = MenuStatus.Started,
+
+                
+            };
+            return userSearchState;
+        }
+        //public void SetStatus(MenuStatus status)
+        //{
+        //    _historyOfMenuStatuses.Add(status);
+        //}
+        public void SetChatId(string chatId)
+        {
+            ChatId = chatId;
+        }
+        public void SetBrand(string brand)
+        {
+            Brand = brand;
+            CurrentMenuStatus = MenuStatus.BrandChosen;
+            _historyOfMenuStatuses.Add(MenuStatus.BrandChosen);
+        }
+
+        public void SetCategory(string category)
+        {
+            Category = category;
+            CurrentMenuStatus = MenuStatus.CategoryChosen;
+            _historyOfMenuStatuses.Add(MenuStatus.CategoryChosen);
+        }
+
+        public void AddStatusToHistory(MenuStatus status)
+        {
+            _historyOfMenuStatuses.Add(status);
+        }
+        //public void SetCurrentState(MenuStatus status)
+        //{
+        //    var currentState = MenuStatus(status);
+        //}
+
+        public void BackOnPreviousStep(MenuStatus status)
+        {
+            _historyOfMenuStatuses.Last();
+            _historyOfMenuStatuses.Remove(status);
+        }
+
+        public void GetNextMenu()
+        {
+            throw new NotImplementedException(); // от того какой будет следующий статус в основном классе TCH зависит показ следующего меню
+            // те мы передаем сюда аргумент со статусом , который потом будем сопоставлять с словарем и присылать в ответ нужное меню
+        }
+
+        public void IsFiniShed()//
+        {
+            throw new NotImplementedException();
+        }
+
+        public UserSearchStateDTO TOMongoRepresintation() //  вызываем этот метод когда все поля уже заполнены и их можно передавать в репозиторий??
```

Comment:
We don't need ALL the fields. All fields - except status and userId are optional